### PR TITLE
Cosmos: update 2019 08 01 preview

### PR DESCRIPTION
**Pull request by ChenTanyi**

### CODE REVIEW

File: azure-mgmt-cosmosdb/src/main/java/com/microsoft/azure/management/cosmosdb/CosmosDBAccount.java [hunk 1]
```diff
@@ -614,5 +675,35 @@
...
+         */
+        interface WithPrivateEndpointConnection {
+            /**
+             * Start the definition of a private endpoint connection to be attached
+             * to the cosmos db account
+             *
+             * @param name the reference name for the private endpoint connection
+             * @return the first stage of a private endpoint connection definition
+             */
+            PrivateEndpointConnection.UpdateDefinitionStages.Blank<WithOptionals> defineNewPrivateEndpointConnection(String name);
```
Comment (weidongxu-microsoft) [inline azure-mgmt-cosmosdb/src/main/java/com/microsoft/azure/management/cosmosdb/CosmosDBAccount.java]: do we need "New"?

Comment (ChenTanyi) [reply azure-mgmt-cosmosdb/src/main/java/com/microsoft/azure/management/cosmosdb/CosmosDBAccount.java]: like VirtualMachine has defineNewExtension

File: azure-mgmt-cosmosdb/src/main/java/com/microsoft/azure/management/cosmosdb/CosmosDBAccount.java [hunk 2]
```diff
@@ -133,6 +134,50 @@
...
+     */
+    Observable<List<PrivateLinkResource>> listPrivateLinkResourcesAsync();
+
+    /**
+     * @return all private link resources in the account.
+     */
+    List<PrivateLinkResource> listPrivateLinkResources();
+
+    /**
+     * @param groupName group name of private link resource
```
Comment (weidongxu-microsoft) [inline azure-mgmt-cosmosdb/src/main/java/com/microsoft/azure/management/cosmosdb/CosmosDBAccount.java]: groupName or groupId or privateLinkName?

Comment (ChenTanyi) [reply azure-mgmt-cosmosdb/src/main/java/com/microsoft/azure/management/cosmosdb/CosmosDBAccount.java]: the api is called groupName https://github.com/Azure/azure-rest-api-specs/blob/de100e16eabbd5d41551cba8b1d8cd97f3701bf9/specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2019-08-01-preview/privateLinkResources.json#L76

Comment (ChenTanyi) [reply azure-mgmt-cosmosdb/src/main/java/com/microsoft/azure/management/cosmosdb/CosmosDBAccount.java]: but the sample response is like groupId https://github.com/Azure/azure-rest-api-specs/blob/de100e16eabbd5d41551cba8b1d8cd97f3701bf9/specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2019-08-01-preview/examples/CosmosDBPrivateLinkResourceGet.json#L16

**Pull request merged**